lib/stdlib/src/supervisor.erl

```diff
@@ -2020,10 +2032,16 @@ supname(self, Mod) -> {self(), Mod};
 supname(N, _)      -> N.
 
 
-default_hibernate_after(simple_one_for_one) ->
+-compile({inline, [hibernate_after_action/1]}).
+hibernate_after_action(#state{hibernate_after_ref = HRef, hibernate_after = HibernateAfter}) ->
+    {timeout, HibernateAfter, HRef}.
```

Comment:
```suggestion
hibernate_after_action(#state{hibernate_after = 0}) ->
    hibernate;
hibernate_after_action(#state{hibernate_after_ref = HRef, hibernate_after = HibernateAfter}) ->
    {timeout, HibernateAfter, HRef}.
```
It might make sense to bypass the timeout if `HibernateAfter` is `0`, since it is clear then that instant hibernation is wanted. The implementation that was chosen in erlang#9287 is to not receive a maybe waiting message but to instantly call `handle_timeout(HibernateAfterRef, State)`, so it boils down to the same thing.

Reply:
I believe I've seen this before and decided to let the gen_server be smart about what a timeout of 0 means. If it boils down to the same thing, why duplicating the logic here, is it necessary?

Reply:
You do have a point :)